### PR TITLE
Split long ansible.builtin.command commands

### PR DESCRIPTION
**Pull request by mpagot**
# Verification

#### 15sp4 Native fencing MSI
 sle-15-SP4-HanaSr-Azure-Byos-x86_64-Build15-SP4_2024-03-01T05:03:10Z-hanasr_azure_test_sapconf_msi@64bit 
- http://openqaworker15.qa.suse.cz/tests/277723 with `|` :green_circle: 
-  http://openqaworker15.qa.suse.cz/tests/277735 with `>-` and `$id`  :green_circle: 

#### 15sp4 Native fencing SPN
 - sle-15-SP4-HanaSr-Azure-Byos-x86_64-Build15-SP4_2024-03-01T05:03:10Z-hanasr_azure_test_sapconf_spn@64bit -> http://openqaworker15.qa.suse.cz/tests/277724 with `|`  :green_circle: 

#### 15sp4 SBD
 - sle-15-SP4-HanaSr-Azure-Byos-x86_64-Build15-SP4_2024-03-01T05:03:10Z-hanasr_azure_test_sapconf_sbd@64bit -> http://openqaworker15.qa.suse.cz/tests/277725 with `|`  :green_circle: 

#### 15sp6 SBD
 - sle-15-SP6-Qesap-Azure-Byos-Ibs-x86_64-BuildLATEST_AZURE_SLE15_6_BYOS_IBS-qesap_azure_uri@az_Standard_E8s_v3 -> http://openqaworker15.qa.suse.cz/tests/277721 with `|`  :green_circle: 

#### 15sp5 SBD
 - sle-15-SP5-Qesap-Azure-Byos-x86_64-BuildLATEST_AZURE_SLE15_5_BYOS-qesap_azure_ansible_roles_test@64bit -> http://openqaworker15.qa.suse.cz/tests/277722 with `|`  :green_circle: 

### HanaSR test
 sle-15-SP4-HanaSr-Aws-Byos-x86_64-Build15-SP4_2024-03-01T05:03:10Z-hanasr_aws_test_saptune@64bit 
 -  http://openqaworker15.qa.suse.cz/tests/277730 with `|` 
 - http://openqaworker15.qa.suse.cz/tests/277736 with `>-` and `$id`  :green_circle:  fails in the same way as without the PR

### qesap regression
 - sle-15-SP5-Qesap-Aws-Byos-x86_64-BuildLATEST_AWS_SLE15_5_BYOS-qesap_aws_sbd_test@64bit -> http://openqaworker15.qa.suse.cz/tests/277731

### qesap regression
 - sle-15-SP5-Qesap-Gcp-Byos-x86_64-BuildLATEST_GCE_SLE15_5_BYOS-qesap_gcp_sbd_test@64bit -> http://openqaworker15.qa.suse.cz/tests/277732

 - sle-15-SP5-Qesap-Gcp-Byos-x86_64-BuildLATEST_GCE_SLE15_5_BYOS-qesap_gcp_sapconf_test@64bit -> http://openqaworker15.qa.suse.cz/tests/277733

### CODE REVIEW

**Review by lpalovsky — APPROVED** [review]
LGTM :+1: 
This is so much better to read.

File: ansible/playbooks/tasks/aws-cluster-hana.yaml [hunk 1]
```diff
@@ -22,44 +23,104 @@
...
 
 - name: Create HANA topology resource
-  ansible.builtin.command: "crm configure primitive rsc_SAPHanaTopology_HDB_{{ sap_hana_install_sid }}{{ sap_hana_install_instance_number }} ocf:suse:SAPHanaTopology operations $id=\"rsc_sap2_HDB_{{ sap_hana_install_sid }}{{ sap_hana_install_instance_number }}-operations\" op monitor interval=\"10\" timeout=\"600\" op start interval=\"0\" timeout=\"600\" op stop interval=\"0\" timeout=\"300\" params SID=\"{{ sap_hana_install_sid }}\" InstanceNumber=\"{{ sap_hana_install_instance_number }}\""
+  ansible.builtin.command:
+    cmd: |
+      crm configure primitive
+      rsc_SAPHanaTopology_HDB_{{ sap_hana_install_sid }}{{ sap_hana_install_instance_number }}
+      ocf:suse:SAPHanaTopology
+      operations
+      $id="rsc_sap2_HDB_{{ sap_hana_install_sid }}{{ sap_hana_install_instance_number }}-operations"
```
Comment (alvarocarvajald) [inline ansible/playbooks/tasks/aws-cluster-hana.yaml:37]: Did you confirm this `$id` is not replaced by anything?

File: ansible/playbooks/tasks/cluster-bootstrap.yaml [hunk 2]
```diff
@@ -219,51 +223,73 @@
...
     - stonith_action != 'off'
     - is_primary
     - cloud_platform_is_aws
 
 - name: Set rsc_defaults resource stickiness [aws/gcp]
-  ansible.builtin.command: crm configure rsc_defaults \$id="rsc-options" resource-stickiness=1000
+  ansible.builtin.command:
+    cmd: |
+      crm configure rsc_defaults
+      \$id="rsc-options"
```
Comment (alvarocarvajald) [inline ansible/playbooks/tasks/cluster-bootstrap.yaml:291]: So is it `\$id` or `$id`? The `$` sign should go into the configuration I believe. See example here: https://documentation.suse.com/sle-ha/12-SP5/html/SLE-HA-all/cha-ha-manual-config.html#sec-ha-config-crm-global

**Review by alvarocarvajald — APPROVED** [review]
LGTM. I do have a question regarding `$id`

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

File: ansible/playbooks/tasks/cluster-hana.yaml [hunk 3]
```diff
@@ -89,19 +151,27 @@
...
 - name: Cleanup if needed
-  # TODO this task has hardcoded SID and instance number values which should be variables!
-  ansible.builtin.command: "crm resource cleanup {{ rsc_SAPHana }}"
+  ansible.builtin.command:
+    cmd: "crm resource cleanup {{ rsc_SAPHana }}"
   when:
     - reg_crm_status.stdout | regex_search('Failed Resource Actions') | trim | length != 0
     - is_primary
+
+# For debug purpose only
```
Comment (mpagot) [inline ansible/playbooks/tasks/cluster-hana.yaml:173]: New code useful to "see" a summary of the configuration at the end

File: ansible/playbooks/tasks/aws-cluster-hana.yaml (hunk 1 above)
Comment (mpagot) [reply ansible/playbooks/tasks/aws-cluster-hana.yaml:37]: yes confirmed by adding a `crm configure show` at the end

File: ansible/playbooks/tasks/cluster-hana.yaml (hunk 3 above)
Comment (alvarocarvajald) [reply ansible/playbooks/tasks/cluster-hana.yaml:173]: Awesome! Does the regex means that it's only printed when there are errors?

File: ansible/playbooks/tasks/cluster-hana.yaml [hunk 4]
```diff
@@ -22,64 +23,125 @@
...
+      rsc_SAPHanaTopology_HDB_{{ sap_hana_install_sid }}{{ sap_hana_install_instance_number }}
+      ocf:suse:SAPHanaTopology
+      operations
+      $id="rsc_sap2_HDB_{{ sap_hana_install_sid }}{{ sap_hana_install_instance_number }}-operations"
+      op monitor interval="10" timeout="600"
+      op start interval="0" timeout="600"
+      op stop interval="0" timeout="300"
+      params
+      SID="{{ sap_hana_install_sid }}"
+      InstanceNumber="{{ sap_hana_install_instance_number }}""
```
Comment (mpagot) [inline ansible/playbooks/tasks/cluster-hana.yaml:43]: Double `""` at the end. Bug.

**Pull request merged**